fix: Connecting to Datasource - Rehaul (#691)
appsmithorg/appsmith-docs: website/docs/core-concepts/connecting-to-data-sources/authentication/authentication-type/basic-authentication.md

When the server receives a request with the Authorization header, it decodes the base64-encoded string and checks the `username` and `password` against its own records. If the credentials are valid, the server allows the request to proceed. If the credentials are invalid or missing, the server returns an **HTTP 401** Unauthorized error to the client.

To share basic authentication details across multiple APIs, you can use the `Basic` authentication mechanism within datasources. To do this:

 <VideoEmbed host="youtube" videoId="SM0K4XkkJgM" title="basic authentication" caption="Basic Authentication"/> 

* Click on the `+` icon next to the datasources section 
* Add an Authenticated API
* Enter the Base URL of the API (example: `https://mock-api.appsmith.com`) and select the authentication type as **Basic**.

The base64-encoded username and password are automatically added to the header of the API request, allowing the API to verify the identity of the user making the request.

## Authentication

![Select Basic from the available Authentication Type](</img/basic-auth.png>)

### Username

> In basic authentication, the username is a string that identifies the user making the request to the API. It's usually a combination of the user's first and last **name**, or it could be a unique identifier such as an **email address** or **employee number**.

Reviewer: As the page talks about Basic Authentication, we need not specify it again for each param:
```suggestion
The username is a string that identifies the user making the request to the API. It is often a combination of the user's first and last name, or it could be a unique identifier like an email address or employee number.
```

### Password

> In basic authentication, the password is a **secret string** that's used to verify the identity of the user making a request to an API. The password is sent as part of the Authorization header in an HTTP request, along with the username, to the server that's hosting the API.

Reviewer: Same as above, no need to specify in basic authentication.

> The server uses the password to verify the identity of the user and determine whether the request should be allowed to proceed. The password is stored in a encrypted form, to protect it from being accessed by unauthorized users.

Reviewer 2: Should we make this line bold : `The password is stored in a encrypted form`

Author: done
